src/didymus/reader.py

```diff
@@ -0,0 +1,62 @@
+# import here
+import numpy as np
+from didymus import pebble as peb
+
+
+# defining code-specific readers here:
+class OpenmcReader():
+    """
+    Class for reading Openmc pebble coordinates from the
+    OpenMC pack_spheres() function into and arrary of
+    distinct Pebble objects.
+    """
+
+    def __init__(self, coord_array):
+        '''
+        Initializes the OpenmcReader.
+
+        Parameters
+        ----------
+        coord_array : array
+            Array of center coordinates, in x, y, and z, from
+            the pack_spheres() function of OpenMC.
+
+        '''
+        self.coord_array = coord_array
+
+    def pebs(self, peb_rad, mat_ids, uniq_ids):
+        '''
+        Creates an array of didymus Pebble objects, using
+        the central coordinates in coord_array.
+
+        Parameters
+        ----------
+        peb_rad : float
+            Radius of a single pebble, with units matching those
+            used to create center coordinates.  Assumes all
+            pebbles are the same size.
+        mat_ids : array of int or str
+            Array containing the mat_ids associated with each pebble
+            center.  While multiple pebbles can share the same mat_id,
+            the length of the mat_ids array should match coord_array.
+        uniq_ids : array of int
+            Array containing the uniq_ids associated with each pebble
+            center.  Each pebble should have a distinct uniq_id.
+
+        Returns
+        -------
+        peb_array : array of Pebble objects
+            Array containing a unique Pebble object for each
+            coordinate provided in coord_array.
+
+        '''
+        peb_array = []
+        for i, coord, in enumerate(self.coord_array):
+            peb_array.append(
+                peb.Pebble(
+                    coord,
+                    peb_rad,
+                    mat_ids[i],
+                    uniq_ids[i]))
+
+        return peb_array
```

Comment:
```suggestion
        peb_array = [Pebble(coord, pebble_radius, mat_ids[i], pebble_ids[i]) for i, coord in enumerate(self.coord_array)]

        return pebble_array
```

List comprehensions are faster. Is every pebble going to be a unique material? Should mat_ids be part of the initialization of the `OpenMCReader`? It seems cumbersome (and error prone) to have users generate id's for each pebble themselves.

Reply:
Each pebble will not necessarily be a unique material,  I think it's more likely that you will have large numbers of the same pebble material (such as having a unique material description for each pass the pebbles have made through the core).  I don't think having a default for the uniq_ids/pebble_ids is a bad thing, though.